systemservices/marketplace/src/tasks/publishPurchase.ts

```diff
@@ -0,0 +1,25 @@
+import { TaskInputs, TaskOutputs } from "mesg-js/lib/service"
+import Web3 from "web3"
+import { extractEventFromLogs } from "../contracts/utils";
+import { Marketplace } from "../contracts/Marketplace";
+import { servicePurchased } from "../contracts/parseEvents";
+
+export default (
+  web3: Web3,
+  marketplace: Marketplace,
+) => async (inputs: TaskInputs, outputs: TaskOutputs): Promise<void> => {
+  try {
+    if (inputs.signedTransactions.length === 2) {
```

Comment:
what happen if `length === 3` ? I agree in the context of us using it it should never happen but nothing stops someone to do it with the api.

I would do it in 2 different ways:

## Strong check
```js
if (inputs.signedTransactions.length > 2) {
  throw new Error("Can only accept maximum signed transactions")
}
// Then now you can play with the index directly
```

## Soft way
```js
const txs = [...inputs.signedTransactions] // Copy of the array just because it's better if you change it
const importantTx = txs.pop()
for (var tx in txs) {
  await web3.eth.sendSignedTransaction(tx)
}
const receipt = await web3.eth.sendSignedTransaction(importantTx)
```

Reply:
I don't understand what your asking.

The original way was like the soft way you are proposing:
```ts
let receipt = null
for (let i=0; i<inputs.signedTransactions.length; i++) {	
  receipt = await web3.eth.sendSignedTransaction(inputs.signedTransactions[i])
}
```

It could even be done with a reduce

Reply:
Not exactly the same.

My version: You are splitting the data and doing 2 different processing (one loop and one result collection:
You version: You are doing the same processing, getting the result but in the end this processing for the first element is kind of ignored and that's confusing. I agree the result is exactly the same, it's just that it's easier to understand.
Basically:
```
extractTransactions()
processTransactionsWhereWeDontNeedResult()
processTransactionWhereWeNeedResult()
```

It's nothing really blocking anyway but can be improved. It's never good to assign a variable if you don't use it especially if replace it right after.

Reply:
Done